rust/src/basis.rs

```diff
@@ -14,21 +14,41 @@
 // software, applications, hardware, advanced system engineering and early
 // testbed platforms, in support of the nation's exascale computing imperative
 use crate::prelude::*;
-use std::ffi::CString;
-use std::fmt;
+
+// -----------------------------------------------------------------------------
+// CeedBasis option
+// -----------------------------------------------------------------------------
+#[derive(Clone, Copy)]
+pub enum BasisOpt<'a> {
+    Some(&'a Basis),
```

Comment:
Maybe the enum should be called `Basis` and use `BasisRaw` or some such for the (not visible to most users) type we now call `Basis`?

Reply:
It looks like the users still make bases as before to me? This looks like its an emun that returns the pointer from the user's basis object or `CEED_BASIS_COLLOCATED`

Reply:
I think `BasisOpt` is more standard Rust for this kind of things, and I agree with @jeremylt that the user will still manipulate `Basis` most of time, the only time he manipulates `BasisOpt` is to access the special values.

Reply:
Sounds good if that's more idiomatic. I was thinking we could simplify the arg types for methods like `Operator::set_field`, but I'm very much still learning the idioms, and the `impl Into<BasisOpt>` technique is fine with me.